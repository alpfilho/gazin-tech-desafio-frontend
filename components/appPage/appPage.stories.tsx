import { ComponentMeta, ComponentStory } from '@storybook/react';
import { AppPage } from './appPage';

export default {
	title: 'Layout/App Page',
	component: AppPage,
	parameters: {
		layout: 'fullscreen'
	}
} as ComponentMeta<typeof AppPage>;

export const Default: ComponentStory<typeof AppPage> = () => (
	<AppPage>
		<h1>Example Content</h1>
	</AppPage>
);
