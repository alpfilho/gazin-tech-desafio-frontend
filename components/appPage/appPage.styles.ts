import styled from 'styled-components';

export const AppPageContainer = styled.main`
	position: relative;
	flex-grow: 1;
	padding-bottom: 10em;
`;
