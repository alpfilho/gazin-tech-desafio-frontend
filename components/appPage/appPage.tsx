import React from 'react';

import { AppPageContainer } from './appPage.styles';

interface AppPageI {
	className?: string;
}

/**
 * Componente estrutural para padronizar a criação de páginas
 */
export const AppPage: React.FC<AppPageI> = ({ children, className }) => {
	return <AppPageContainer className={className}>{children}</AppPageContainer>;
};
