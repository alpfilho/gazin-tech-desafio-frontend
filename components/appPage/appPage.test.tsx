import { render, screen } from '@testing-library/react';
import { composeStories } from '@storybook/testing-react';

import * as stories from './appPage.stories';

const { Default } = composeStories(stories);

describe('AppPage', () => {
	it('render a app page with an example title', () => {
		render(<Default />);

		const exampleText = screen.getByText('Example Content');

		expect(exampleText).toBeInTheDocument();
		expect(exampleText).toBeVisible();
	});

	it('render AppPage unchanged', () => {
		const { container } = render(<Default />);

		expect(container).toMatchSnapshot();
	});
});
