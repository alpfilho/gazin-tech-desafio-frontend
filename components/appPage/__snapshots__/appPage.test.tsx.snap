// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AppPage render AppPage unchanged 1`] = `
<div>
  <main
    class="AppPageContainer-sc-wxajy8-0 Tyabz"
  >
    <h1>
      Example Content
    </h1>
  </main>
</div>
`;
